uint32 queryId
---
bool success